[docs] Add Expo Go version mismatch troubleshooting (#48255)
expo/expo: docs/pages/troubleshooting/expo-go-version-mismatch.mdx

---
title: '"Project is incompatible with this version of Expo Go" error'
sidebar_title: Expo Go version mismatch
description: Learn why an Expo project is incompatible with the installed version of Expo Go and how to install a compatible version.
---

Expo Go displays this error when the Expo SDK version in a project does not match the SDK version included in the installed Expo Go app:

```text
Project is incompatible with this version of Expo Go
```

You may also see one of these messages:

```text
The project you requested requires a newer version of Expo Go.
This project requires a newer version of Expo Go.
```

## Why this happens

> Each build of Expo Go includes one Expo SDK version. The project's SDK version is determined by the version of the `expo` package in **package.json**. The project and Expo Go SDK versions must match.

Reviewer: ✏️ **style** `prose.md#voice-and-tone`

"is determined by" is passive. "The `expo` package version in **package.json** sets the project's SDK version" says the same thing actively and in fewer words.

```suggestion
Each build of Expo Go includes one Expo SDK version. The `expo` package version in **package.json** sets the project's SDK version. The project and Expo Go SDK versions must match.
```

<!-- docs-review -->

> The Apple App Store and Google Play Store versions of Expo Go can lag behind a newly released SDK. If your project uses the new SDK, the store may not offer a compatible Expo Go build yet.

Reviewer: ⚠️ **correctness** `prose.md#core-principles`

SDK 55 and later are not on the Apple App Store at all, and there is no timeline for them appearing ([changelog](https://expo.dev/changelog/expo-go-and-app-store-may-2026)). "can lag behind" and "may not offer a compatible Expo Go build yet" read as a delay that clears on its own, so an iOS reader could reasonably decide to wait it out. State that the App Store build stops at SDK 54.

```suggestion
Expo Go on the Apple App Store stops at SDK 54, and SDK 55 and later are not available there. The Google Play version can also lag behind a new SDK release. If your project uses a newer SDK than the store build, use one of the options below instead of waiting for a store update.
```

<!-- docs-review -->

## Install a compatible version of Expo Go

Use the instructions for the device where you are testing your project.

### Physical iPhone or iPad

For projects using SDK 55 or later, open [sign.expo.dev](https://sign.expo.dev/), select the SDK version used by your project, and follow the steps to install that Expo Go build on your device.

This installer uses your Apple ID's free developer provisioning, so it does not require a paid Apple Developer Program membership. The certificate is valid for about seven days. Return to [sign.expo.dev](https://sign.expo.dev/) to re-sign and reinstall Expo Go when the certificate expires.

> For projects using SDK 54 or earlier, you cannot install an older version of Expo Go on a physical iOS device. Upgrade the project, use an Android device or iOS Simulator, or create a development build.

Reviewer: ⚠️ **correctness** `review-ai-text.md#t7-overclaiming`

SDK 54 is the build currently on the App Store, and [expo.dev/go](https://expo.dev/go?sdkVersion=54&platform=ios&device=true) offers `npx eas-cli@latest go --sdk-version 54.0.0` for physical iOS devices. SDK 53 is the first version whose page says "Not available", so the cutoff belongs there. [`eas go`](https://expo.fyi/deploy-expo-go-testflight) builds Expo Go for a chosen SDK and ships it to your TestFlight internal team, and this page never mentions it.

```suggestion
For projects using SDK 54, install Expo Go from the Apple App Store. You can also use [`eas go`](https://expo.fyi/deploy-expo-go-testflight) to build Expo Go for SDK 54 or later and distribute it through your TestFlight internal team, which needs an Apple Developer Program membership.

For projects using SDK 53 or earlier, you cannot install an older version of Expo Go on a physical iOS device. Upgrade the project, use an Android device or iOS Simulator, or create a development build.
```

<!-- docs-review -->

### Android device, Android Emulator, or iOS Simulator

Visit [expo.dev/go](https://expo.dev/go), select the SDK version used by your project and your target platform, then install the compatible Expo Go build.

You can also use the [`expo-go` CLI](/develop/tools/#expo-go-cli) to download a specific version.

After installing the compatible build, restart the development server with `npx expo start` and open the project again.

> ## If the versions should already match

Reviewer: 💡 **nit** `links-and-headings.md#titles-and-headings`

This is the one H2 on the page that is not a verb phrase, so it breaks the scan against "Install a compatible version of Expo Go" and "Upgrade the project or use a development build".

```suggestion
## Check the project's SDK version
```

<!-- docs-review -->

If the error remains after installing the correct Expo Go build:

1. Check the `expo` dependency in **package.json** to confirm the project's SDK version.

> 2. If **app.json** contains an `sdkVersion` field, remove it or make sure that it matches the `expo` dependency.

Reviewer: ✏️ **style** `terminology.md#expo-product-and-feature-names`

`sdkVersion` is an app config field, so a project using **app.config.js** or **app.config.ts** hits the same problem and skips this step. Use "app config" linked to the configuration page.

```suggestion
2. If your [app config](/workflow/configuration/) contains an `sdkVersion` field, remove it or make sure that it matches the `expo` dependency.
```

<!-- docs-review -->
